File: Grokking Algorithms/01_introduction_to_algorithms/Binary Search/BinarySearch.java
import java.lang.Math;

public class BinarySearch {
    public static void main(String[] args) {
        int[] list = {1, 3, 5, 7, 9};

        System.out.println(BinarySearch(list, 3));
    }

    public static int BinarySearch(int[] list, int number) {
        int min = 0;
        int max = list.length - 1;

        while(min <= max) {
            int mid = Math.round((min + max) / 2);
            int guess = list[mid];

            if(guess == number){
                return mid;
            }

            if (guess > number) {
                max = mid - 1;
            } else {
                min = mid + 1;
            }
        }

        return 0;
    }
}

File: Grokking Algorithms/01_introduction_to_algorithms/Binary Search/binary_search.py
def binary_search(list, item):

    min = 0
    max = len(list) - 1

    while min <= max:
        mid = (min + max) // 2
        guess = list[mid]

        if guess == item:
            return mid

        if guess > item:
            max = mid - 1
        else:
            min = mid + 1

    return None


my_list = [15, 27, 39, 77, 92, 108, 121, 135, 200, 432, 12, 8, 1, 0]
my_list.sort()

print(my_list)
print(binary_search(my_list, 77))
